src/nvim/eval.c

```diff
@@ -1754,10 +1754,10 @@ ex_let_one (
           vim_setenv(name, p);
           if (STRICMP(name, "HOME") == 0)
             init_homedir();
-          else if (didset_vim && STRICMP(name, "VIM") == 0)
+          else if (didset_vim && (STRICMP(name, "VIM") == 0 || STRICMP(name, "NVIM") == 0))
```

Comment:
Honestly I was afraid to change it, since I have no idea why it is set to STRICMP.